### PR TITLE
Support config reactive datasource with list of database urls for fault tolerance and load balance

### PR DESCRIPTION
**Pull request by benstonezhang**
Add support to config load balance urls for reactive datasources in application.properties, for example:

```
quarkus.datasource.reactive.url=mysql://host1:3306/dbname,mysql://host2:3306/dbname,mysql://host3:3306/dbname
```

or

```
quarkus.datasource.reactive.url[0]=mysql://host1:3306/dbname
quarkus.datasource.reactive.url[1]=mysql://host2:3306/dbname
quarkus.datasource.reactive.url[2]=mysql://host3:3306/dbname
```

The original PR is [#31852](https://github.com/quarkusio/quarkus/pull/31852), which only update reactive mysql client. As @tsegismont suggested, I replaced `public Optional<String> url` with `public Optional<List<String>> urls` in DataSourceReactiveRuntimeConfig, and all of reactive database client were updated accordingly.

There is one exception: Vert.x oracle client already have build-in load balance support, and there is no such function like `OraclePool.pool(Vertx vertx, List<OracleConnectOptions> databases, PoolOptions options)`. So only the first url is valid and others is discarded even if you config more urls.

The original PR title maybe improperly and misleading, so I create this one.

### CODE REVIEW

**Review by tsegismont — COMMENTED** [review]
Thank you @benstonezhang

File: extensions/reactive-mysql-client/deployment/src/test/java/io/quarkus/reactive/mysql/client/MySQLPoolLoadBalanceTest.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import org.hamcrest.Matchers;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusDevModeTest;
+import io.restassured.RestAssured;
+
+public class MySQLPoolLoadBalanceTest {
```
[the remark this answers is not in the document]
Comment (benstonezhang) [reply extensions/reactive-mysql-client/deployment/src/test/java/io/quarkus/reactive/mysql/client/MySQLPoolLoadBalanceTest.java:11]: revised, please check.
For DB2 and SQL Server, as the  [Reactive DB2 Client](https://vertx.io/docs/vertx-db2-client/java/#_advanced_pool_configuration) and [Reactive MSSQL Client](https://vertx.io/docs/vertx-mssql-client/java/#_advanced_pool_configuration) documented, Vert.x do support advanced pool configuration:
```
DB2Pool pool = DB2Pool.pool(Arrays.asList(server1, server2, server3), options);
MSSQLPool pool = MSSQLPool.pool(Arrays.asList(server1, server2, server3), options);
```
so I think we should support it too.

File: extensions/reactive-oracle-client/runtime/src/main/java/io/quarkus/reactive/oracle/client/runtime/OraclePoolRecorder.java [hunk 2]
```diff
@@ -112,7 +113,11 @@ private OracleConnectOptions toOracleConnectOptions(DataSourceRuntimeConfig data
             DataSourceReactiveOracleConfig dataSourceReactiveOracleConfig) {
         OracleConnectOptions oracleConnectOptions;
         if (dataSourceReactiveRuntimeConfig.url.isPresent()) {
-            String url = dataSourceReactiveRuntimeConfig.url.get().get(0);
+            List<String> urls = dataSourceReactiveRuntimeConfig.url.get();
+            if (urls.size() > 1) {
+                log.warn("The Oracle driver does not support multiple URLs. This first is used, and others is ignored.");
```
Comment (tsegismont) [inline extensions/reactive-oracle-client/runtime/src/main/java/io/quarkus/reactive/oracle/client/runtime/OraclePoolRecorder.java:118]:
```suggestion
                log.warn("The Reactive Oracle Client does not support multiple URLs. The first one will be used, and others will be ignored.");
```

**Review by tsegismont — COMMENTED** [review]
Thank you for the updates. Yes, in the Vert.x API / documentation it is possible but, in practice, I'm aware of multi-master systems only for MySQL, Postgres and Oracle.

For Oracle we can't support it the same way because the driver has to take care of it.

**Comment by benstonezhang** [comment]
I'm not family with DB2 and SQL Server. I will rework it tomorrow.

**Comment by benstonezhang** [comment]
@tsegismont code updated, please check. Thx.

**Comment by tsegismont** [comment]
Can you please rebase your PR rather than merging the `main` branch into your branch?

**Comment by tsegismont** [comment]
It doesn't look good @benstonezhang

**Comment by benstonezhang** [comment]
just done rebase and push

**Comment by benstonezhang** [comment]
@tsegismont Sorry, seems I had messed this branch. I create a new PR [#32128](https://github.com/quarkusio/quarkus/pull/32128), please check.

**Pull request closed without merging**